Mlem/API/Requests/Comment/GetCommentsRequest.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  GetCommentsRequest.swift
+//  Mlem
+//
+//  Created by Nicholas Lawson on 08/06/2023.
+//
+
+import Foundation
+
+struct GetCommentsRequest: APIGetRequest {
+
+    typealias Response = GetCommentsResponse
+
+    let instanceURL: URL
+    let path = "comment/list"
+    let queryItems: [URLQueryItem]
+
+    // lemmy_api_common::comment::GetComments
+    init(
+        account: SavedAccount,
+        postId: Int,
+        maxDepth: Int = 15,
+        type: FeedType = .all,
+        // TODO replace this with proper comment sort type, current one has .active, which is not in API
+        sort: String? = nil,
+        page: Int? = nil,
+        limit: Int? = nil,
+        communityId: Int? = nil,
+        communityName: String? = nil,
+        parentId: Int? = nil,
+        savedOnly: Bool? = nil
+    ) {
+        self.instanceURL = account.instanceLink
+        self.queryItems = [
+            .init(name: "auth", value: account.accessToken),
+            .init(name: "post_id", value: "\(postId)"),
+            .init(name: "max_depth", value: "\(maxDepth)"),
+            .init(name: "type_", value: type.rawValue),
+            .init(name: "sort", value: sort),
+            .init(name: "page", value: page.map(String.init)),
```

Comment:
why for some of the values do we need to add the `.map`, but for other optional arguments like `sort` above we don't?

some of these arguments were not present before, which suggests they can be optional and not sent, if that were the case given the filtering done later, could they not all be simplified to just:
```swift
.init(name: "name", value: value)
```

Reply:
oh hang on, I'm being silly - it's because they all need coerced into `String` for the types which aren't 🙈 

carry on